### PR TITLE
Add test cases for noImplicitReturns to testing suite

### LINKED ISSUE #19: Create tests for noImplicitReturns

### PR DESCRIPTION
**Pull request by guor8lei**
Fixes #19 

Integrates the test cases from `no_implicit_returns_sample.ts` into the testing suite which automatically runs the CLI and compares the expected output to the actual output.

Compares outputs by ASTs, to ensure that syntactically the two files are the same.

Split test cases from one large file into separate files for different cases.

### CODE REVIEW

File: test/no_implicit_returns_test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    project.addSourceFilesAtPaths(inputFilePaths);
+    project.resolveSourceFileDependencies();
+
+    new Runner(
+      /* args*/ undefined,
+      project,
+      /* parser */ undefined,
+      /* errorDetector */ undefined,
+      /* manipulators */ undefined,
```
Comment (jiady) [inline test/no_implicit_returns_test.ts:60]: shall we construct one manipulator here?

File: test/test_files/no_implicit_returns/empty_return.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+function noValueReturnInIf(name: string) {
+    if (name === 'Bob') {
+      return;
```
Comment (jiady) [inline test/test_files/no_implicit_returns/empty_return.ts:3]: let's add a comment mentioning this line need to fix.

File: test/test_files/no_implicit_returns/empty_return.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    }
+    return 'Hello Not-Bob';
+  }
+  
+  function noExplitReturnUndefined(name: string) {
+    if (name === 'Bob') {
+      return 'Hello Bob';
+    }
+    return;
```
Comment (jiady) [inline test/test_files/no_implicit_returns/empty_return.ts:12]: let's add a comment mentioning this line need to fix.

File: test/test_files/no_implicit_returns/no_return.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+function noEndValueReturn(name: string) {
+    if (name === 'Bob') {
+      return 'Hello Bob';
+    }
```
Comment (jiady) [inline test/test_files/no_implicit_returns/no_return.ts:4]: let's add a comment mentioning this line need to fix.

File: test/test_files/golden/no_implicit_returns/no_return.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+function noEndValueReturn(name: string) {
+  if (name === 'Bob') {
+    return 'Hello Bob';
+  }
+  // typescript-flag-upgrade automated fix: --noImplicitReturns
+  return undefined;
+}
```
Comment (michaelyara) [inline test/test_files/golden/no_implicit_returns/no_return.ts:7]: Nit: new line at end of file

File: test/test_files/no_implicit_returns/empty_return.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  }
+
```
Comment (michaelyara) [inline test/test_files/no_implicit_returns/empty_return.ts:14]: newline

File: test/no_implicit_returns_test.ts (hunk 1 above)
Comment (guor8lei) [reply test/no_implicit_returns_test.ts]: Fixed

**Pull request merged**